test/unit-test/downloader_utest.c

```diff
@@ -301,13 +301,121 @@ void test_processReceivedDataBlock_processesJSONBlock( void )
 
     uint8_t decodedData[ mqttFileDownloader_CONFIG_BLOCK_SIZE ];
     size_t dataLength = 0;
+    int32_t fileId = 0;
+    int32_t blockId = 0;
+    int32_t blockSize = 0;
+    const char * message = "{\"f\": \"12\", \"i\": \"1\", \"l\": \"512\", \"p\": \"dGVzdA==\"}";
 
-    bool result = mqttDownloader_processReceivedDataBlock( &context, ( uint8_t * ) "{\"p\": \"dGVzdA==\"}", strlen( "{\"p\": \"dGVzdA==\"}" ), decodedData, &dataLength );
+    bool result = mqttDownloader_processReceivedDataBlock( &context, ( uint8_t * ) message, strlen( message ), &fileId, &blockId, &blockSize, decodedData, &dataLength );
 
     TEST_ASSERT_TRUE( result );
+    TEST_ASSERT_EQUAL( 12, fileId );
+    TEST_ASSERT_EQUAL( 1, blockId );
+    TEST_ASSERT_EQUAL( 512, blockSize );
     TEST_ASSERT_EQUAL( 4, dataLength );
 }
 
+void test_processReceivedDataBlock__invalidJSONBlock_fileIdNotANumber( void )
+{
+    MqttFileDownloaderContext_t context = { 0 };
+
+    context.dataType = DATA_TYPE_JSON;
+
+    uint8_t decodedData[ mqttFileDownloader_CONFIG_BLOCK_SIZE ];
+    size_t dataLength = 0;
+    int32_t fileId = 0;
+    int32_t blockId = 0;
+    int32_t blockSize = 0;
+    const char * message = "{\"f\": \"file\", \"i\": \"1\", \"l\": \"512\", \"p\": \"dGVzdA==\"}";
+
+    MQTTFileDownloaderStatus_t result = mqttDownloader_processReceivedDataBlock( &context, ( uint8_t * ) message, strlen( message ), &fileId, &blockId, &blockSize, decodedData, &dataLength );
+
+    TEST_ASSERT_EQUAL( 7, result );
+    TEST_ASSERT_EQUAL( 0, fileId );
+    TEST_ASSERT_EQUAL( 0, blockId );
+    TEST_ASSERT_EQUAL( 0, blockSize );
+    TEST_ASSERT_EQUAL( 0, dataLength );
+}
+
+void test_processReceivedDataBlock__invalidJSONBlock_blockIdNotANumber( void )
+{
+    MqttFileDownloaderContext_t context = { 0 };
+
+    context.dataType = DATA_TYPE_JSON;
+
+    uint8_t decodedData[ mqttFileDownloader_CONFIG_BLOCK_SIZE ];
+    size_t dataLength = 0;
+    int32_t fileId = 0;
+    int32_t blockId = 0;
+    int32_t blockSize = 0;
+    const char * message = "{\"f\": \"0\", \"i\": \"block\", \"l\": \"512\", \"p\": \"dGVzdA==\"}";
+
+    MQTTFileDownloaderStatus_t result = mqttDownloader_processReceivedDataBlock( &context, ( uint8_t * ) message, strlen( message ), &fileId, &blockId, &blockSize, decodedData, &dataLength );
+
+    TEST_ASSERT_EQUAL( 7, result );
+    TEST_ASSERT_EQUAL( 0, fileId );
+    TEST_ASSERT_EQUAL( 0, blockId );
+    TEST_ASSERT_EQUAL( 0, blockSize );
+    TEST_ASSERT_EQUAL( 0, dataLength );
+}
+
+void test_processReceivedDataBlock__invalidJSONBlock_blockSizeNotANumber( void )
```

Comment:
Nit: I know this doesn't affect coverage metrics, but a `blockSizeIsNegative` would be a nice to have test to show that negative block sizes aren't supported

Reply:
Updated in https://github.com/aws/aws-iot-core-mqtt-file-streams-embedded-c/pull/23/commits/ec66b1d9c00b8b387696b131fc01ed5ee73f1910.